src/utils/json-data.js

```diff
@@ -21,21 +21,20 @@ const getGlobalData = () => {
 
 const buildGlobalDataFromJson = () => {
   const root = shared.resolveAssetsPath('data');
-  const filePaths = glob.sync(path.join(root, '*.json'));
+  const filePaths = glob.sync(path.join(root, '/**/*.json'));
 
   if (!filePaths.length) {
     return;
   }
 
   const dataObject = filePaths.reduce((acc, filePath) => {
-    const fileName = path.basename(filePath);
-    const propertyName = convertFileNameToObjectPropertyName(fileName);
-
-    if (!isPropertyNameValid(propertyName)) {
+    const fileName = filePath.replace(root, '');
+    const propertyNameSegments = convertFileNameToObjectPropertyNameSegments(fileName);
+    if (!isPropertyNameValid(propertyNameSegments)) {
       console.error(
         new shared.StachePluginError([
           `A valid Object property could not be determined from file ${fileName}!`,
-          `The property key '${propertyName}' cannot be used. Please choose another file name.`
+          `The property keys '${propertyNameSegments}' cannot be used. Please rename your file and path.`
```

Comment:
I'm assuming the console will attempt to flatten the array, but should the `propertyNameSegments` be joined to make sure it's always shown correctly?